### PR TITLE
fix: [CDS-54878]: fixed issue with http url in webhook

### PR DESCRIPTION
**Pull request by devkimittal**
## Describe your changes
This pr fixes https://harness.atlassian.net/browse/CI-6869 as well

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45067)

### CODE REVIEW

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/eventmapper/filters/impl/GitWebhookTriggerRepoFilter.java [hunk 1]
```diff
@@ -139,6 +139,7 @@ HashSet<String> getUrls(Repository repository, String sourceRepoType) {
     HashSet<String> urls = new HashSet<>();
 
     String httpUrl = repository.getHttpURL().toLowerCase();
+    httpUrl = convertToHttps(httpUrl);
```
Comment (sribalij) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/eventmapper/filters/impl/GitWebhookTriggerRepoFilter.java:142]: This is already being done in `io.harness.ngtriggers.eventmapper.filters.impl.GitWebhookTriggerRepoFilter#evaluateWrapperForRepoLevelGitConnector`
`io.harness.ngtriggers.eventmapper.filters.impl.GitWebhookTriggerRepoFilter#evaluateWrapperForProjectLevelGitConnector`
`io.harness.ngtriggers.eventmapper.filters.impl.GitWebhookTriggerRepoFilter#evaluateWrapperForAccountLevelGitConnector`

Do we need to remove from there?

Comment (devkimittal) [reply pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/eventmapper/filters/impl/GitWebhookTriggerRepoFilter.java:142]: No, the problem here is we are comparing git url from connector to the urls we get from the webhook. The logic that I wrote in all the 3 methods you mentioned, we assumed that we will receive the https url from the webhook but in the ticket mentioned we are receiving the http url.
Basically 4 cases are there
1. http from webhook and https from connector
2. https from webhook and https from connector
3. http from webhook and http from connector
4. https from webhook and http from connector

The logic written in three methods will only solve case 4th, 2nd no handling required. this pr will solve case 1 and 3.

Comment (sribalij) [reply pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/eventmapper/filters/impl/GitWebhookTriggerRepoFilter.java:142]: Got it. Saw this issue happen on https://harness.atlassian.net/browse/CDS-54878

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45067#-:-NQ_4mPu8FJiahnOjdIc:bnfp4nl)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by raghavharness — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45067#-:-NQ_iOjqDhWkxrnD5eWM:bnfp4nl)* status: 6 file(s) to review. 1 of 2 approvals obtained.

**Review by sribalij — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 4 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45067)* status: :shipit: complete! ✓

**Comment by devkimittal** [comment]
trigger messagecheck

**Pull request merged**